rpcs3/Emu/NP/rpcn_client.cpp

```diff
@@ -2192,19 +2192,298 @@ namespace rpcn
 		return forge_request_with_com_id(builder, pr_com_id, CommandType::SetPresence, rpcn_request_counter.fetch_add(1));
 	}
 
+	bool rpcn_client::createjoin_room_gui(u32 req_id, const SceNpCommunicationId& communication_id, const SceNpMatchingAttr* attr_list)
+	{
+		flatbuffers::FlatBufferBuilder builder(1024);
+
+		u32 total_slots = 0;
+		u32 private_slots = 0;
+		bool privilege_grant = false;
+		bool stealth = false;
+
+		std::vector<flatbuffers::Offset<MatchingAttr>> vec_attrs;
+
+		for (const SceNpMatchingAttr* cur_attr = attr_list; cur_attr != nullptr; cur_attr = cur_attr->next ? cur_attr->next.get_ptr() : nullptr)
+		{
+			ensure(cur_attr->type == SCE_NP_MATCHING_ATTR_TYPE_BASIC_NUM || cur_attr->type == SCE_NP_MATCHING_ATTR_TYPE_GAME_BIN ||
+					   cur_attr->type == SCE_NP_MATCHING_ATTR_TYPE_GAME_NUM,
+				"Invalid attribute type in createjoin_room_gui");
+
+			switch (cur_attr->type)
+			{
+			case SCE_NP_MATCHING_ATTR_TYPE_BASIC_NUM:
+			{
+				ensure(cur_attr->id == SCE_NP_MATCHING_ROOM_ATTR_ID_TOTAL_SLOT || cur_attr->id == SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVATE_SLOT ||
+						   cur_attr->id == SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVILEGE_TYPE || cur_attr->id == SCE_NP_MATCHING_ROOM_ATTR_ID_ROOM_SEARCH_FLAG,
+					"Invalid basic num attribute id");
+
+				switch (cur_attr->id)
+				{
+				case SCE_NP_MATCHING_ROOM_ATTR_ID_TOTAL_SLOT:
+					total_slots = cur_attr->value.num;
+					break;
+				case SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVATE_SLOT:
+					private_slots = cur_attr->value.num;
+					break;
+				case SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVILEGE_TYPE:
+					ensure(cur_attr->value.num == SCE_NP_MATCHING_ROOM_PRIVILEGE_TYPE_NO_AUTO_GRANT || cur_attr->value.num == SCE_NP_MATCHING_ROOM_PRIVILEGE_TYPE_AUTO_GRANT, "Invalid SCE_NP_MATCHING_ROOM_ATTR_ID_PRIVILEGE_TYPE value");
+					privilege_grant = (cur_attr->value.num == SCE_NP_MATCHING_ROOM_PRIVILEGE_TYPE_AUTO_GRANT);
+					break;
+				case SCE_NP_MATCHING_ROOM_ATTR_ID_ROOM_SEARCH_FLAG:
+					ensure(cur_attr->value.num == SCE_NP_MATCHING_ROOM_SEARCH_FLAG_OPEN || cur_attr->value.num == SCE_NP_MATCHING_ROOM_SEARCH_FLAG_STEALTH, "Invalid SCE_NP_MATCHING_ROOM_ATTR_ID_ROOM_SEARCH_FLAG value");
+					stealth = (cur_attr->value.num == SCE_NP_MATCHING_ROOM_SEARCH_FLAG_STEALTH);
+					break;
+				default:
+					fmt::throw_exception("Unreachable");
```

Comment:
Remove ensure above and do
```suggestion
					fmt::throw_exception("Invalid basic num attribute id");
```